core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoFacade.java

```diff
@@ -0,0 +1,317 @@
+import android.os.Handler;
+import android.os.Looper;
+import android.view.Surface;
+
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.DefaultLoadControl;
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.ExoPlayerFactory;
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.decoder.DecoderCounters;
+import com.google.android.exoplayer2.source.AdaptiveMediaSourceEventListener;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.google.android.exoplayer2.ui.SimpleExoPlayerView;
+import com.google.android.exoplayer2.upstream.DataSpec;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.video.VideoRendererEventListener;
+import com.novoda.noplayer.ContentType;
+import com.novoda.noplayer.listeners.BitrateChangedListeners;
+import com.novoda.noplayer.listeners.InfoListeners;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class ExoPlayerTwoFacade implements VideoRendererEventListener {
+
+    private int videoWidth;
+    private int videoHeight;
+
+    private final SimpleExoPlayer exoPlayer;
+    private final MediaSourceFactory mediaSourceFactory;
+
+    private List<Listener> listeners = new CopyOnWriteArrayList<>();
+    private InfoListeners infoListeners;
+    private BitrateChangedListeners bitrateChangedListeners;
+    private InternalErrorListener internalErrorListener;
+
+    public static ExoPlayerTwoFacade newInstance(Context context) {
+        DefaultDataSourceFactory defaultDataSourceFactory = new DefaultDataSourceFactory(context, "user-agent");
+        Handler handler = new Handler(Looper.getMainLooper());
+        DefaultTrackSelector trackSelector = new DefaultTrackSelector();
+        SimpleExoPlayer exoPlayer = ExoPlayerFactory.newSimpleInstance(context, trackSelector, new DefaultLoadControl());
+        MediaSourceFactory mediaSourceFactory = new MediaSourceFactory(defaultDataSourceFactory, handler);
+        return new ExoPlayerTwoFacade(exoPlayer, mediaSourceFactory);
+    }
+
+    public ExoPlayerTwoFacade(SimpleExoPlayer exoPlayer, MediaSourceFactory mediaSourceFactory) {
+        this.exoPlayer = exoPlayer;
+        this.mediaSourceFactory = mediaSourceFactory;
+    }
+
+    public boolean getPlayWhenReady() {
+        if (hasPlayer()) {
+            return exoPlayer.getPlayWhenReady();
+        } else {
+            return false;
```

Comment:
Not really. Most of the logging is done in `ExoPlayer.EventListener` which we then forward to the attached listeners.